apps/lib/publisher_agent/config.go

```diff
@@ -86,11 +84,18 @@ func LoadConfig(configFilePath string, keysFilePath string) (*StorkPublisherAgen
 	}
 
 	var keysFile KeysFile
-	err = json.Unmarshal(keyFileData, &keysFile)
-	if err != nil {
-		return nil, fmt.Errorf("failed to unmarshal keys file: %w", err)
+
+	// only deserialize keyFileData if keysFilePath was successfully read
+	if keyFileReadErr == nil {
+		err = json.Unmarshal(keyFileData, &keysFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal keys file: %w", err)
+		}
 	}
 
+	// overwrite keysFile with any set env vars
+	loadKeysFileFromEnv(&keysFile)
```

Comment:
I generally don't like the pattern of passing a variable to a static function and having the function mutate that variable in place, since someone reading this code might not expect this variable to change after this call.

What do you think about making loadKeysFileFromEnv a method of the `KeysFile` object, so we would call something like `keysFile.updateFromEnvVars()` to make it a little more obvious that we're changing the object.

Reply:
Agreed, changed.